src/Sarif/Writers/SarifLogger.cs

```diff
@@ -241,14 +243,24 @@ public void AnalysisStopped(RuntimeConditions runtimeConditions)
 
         public void Log(IRule rule, Result result)
         {
+            if (rule.Id != result.RuleId)
+            {
+                //The rule id '{0}' specified by the result does not match the actual id of the rule '{1}'
+                string message = string.Format(SdkResources.ResultRuleIdDoesNotMatchRule,
+                    result.RuleId.ToString(),
+                    rule.Id.ToString());
```

Comment:
? `string.Format` calls `ToString()` on its arguments automatically (and it uses whatever `string.Format` was called with, in this case `CurrentCulture` by default (although FxCop thinks, and I agree, that you should specify it explicitly). So this works:

```
string message = string.Format(
                    CultureInfo.CurrentCulture,
                    SdkResources.ResultRuleIdDoesNotMatchRule,
                    result.RuleId,
                    rule.Id);
```

... whether or not the arguments happen to be strings. Am I misunderstanding you?